xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/internal/html/DefaultHTMLCleanerTest.java

```diff
@@ -518,16 +530,30 @@ public void controlCharacters() throws Exception
         htmlInput = "<p>&#8;</p>";
         document = this.cleaner.clean(new StringReader(htmlInput));
 
+        // HtmlCleaner currently doesn't handle properly unicode characters: asking it to recognize them
+        // involves that all entities will be escaped during the parsing and that's not what we want. So we
+        // keep them encoded.
         textContent =
             document.getElementsByTagName("p").item(0).getTextContent();
         assertEquals("&#8;", textContent);
```

Comment:
I would make more explicit that this behavior is insane with a TODO and an issue on HTMLCleanr side.